src/goto-programs/show_goto_functions_xml.cpp

```diff
@@ -0,0 +1,146 @@
+                     x=y</instructionValue>
+                </instruction>
+              </instructions>
+            </function>
+          </functions>
+
+\*******************************************************************/
+
+xmlt show_goto_functions_xmlt::get_goto_functions(
+  const goto_functionst &goto_functions)
+{
+  xmlt xml_functions=xmlt("functions");
+  for(const auto &function_entry : goto_functions.function_map)
+  {
+    const irep_idt &function_name=function_entry.first;
+    const goto_functionst::goto_functiont &function=function_entry.second;
+
+    xmlt &xml_function=xml_functions.new_element("function");
+    xml_function.set_attribute("name", id2string(function_name));
+    xml_function.set_attribute_bool(
+      "isBodyAvailable", function.body_available());
+    bool is_internal=(has_prefix(id2string(function_name), CPROVER_PREFIX) ||
+                      function_name==ID__start);
+    xml_function.set_attribute_bool("isInternal", is_internal);
+
+    if(function.body_available())
+    {
+      xmlt &xml_instructions=xml_function.new_element("instructions");
+      for(const goto_programt::instructiont &instruction :
+        function.body.instructions)
+      {
+        xmlt &instruction_entry=xml_instructions.new_element("instruction");
+
+        std::ostringstream instruction_id_builder;
+        instruction_id_builder << instruction.type;
+
+        instruction_entry.set_attribute(
+          "instructionId", instruction_id_builder.str());
+
+        if(instruction.code.source_location().is_not_nil())
+        {
+          instruction_entry.new_element(
+            xml(instruction.code.source_location()));
+        }
+
+        std::ostringstream instruction_builder;
+        function.body.output_instruction(
+          ns, function_name, instruction_builder, instruction);
+
+        xmlt &instruction_value=
+          instruction_entry.new_element("instructionValue");
+        instruction_value.data=instruction_builder.str();
+        instruction_value.elements.clear();
+      }
+    }
+  }
+  return xml_functions;
+}
+
+/*******************************************************************\
+
+Function: show_goto_functions_xmlt::print_goto_functions
+
+  Inputs:
+   goto_functions - the goto functions that make up the program
+   out - the stream to write the object to
+   append - should a command and newline be appended to the stream
+            before writing the xml object. Defaults to true
+
+ Outputs:
+
+ Purpose: Print the xml object generated by
+          show_goto_functions_xmlt::show_goto_functions to the provided
+          stream (e.g. std::cout)
+
+\*******************************************************************/
+
+void show_goto_functions_xmlt::print_goto_functions(
+  const goto_functionst &goto_functions,
+  std::ostream &out,
+  bool append /*=true*/)
+{
+  const xmlt &function_output=get_goto_functions(goto_functions);
+  if(append)
+  {
+    out << "\n";
```

Comment:
I did this to keep consistency with the json class, but (I think) it adds a superfluous line, so could call with `append=false` or could drop this whole thing.